src/io.rs

```diff
@@ -225,11 +235,28 @@ pub trait Read<'a> {
     fn remaining_data(&self) -> usize;
 
     /// Reads a little-endian integer.
-    fn read_le<I: LeInt>(&mut self) -> Result<I, Error> {
+    fn read_le<I: LeInt>(&mut self) -> Result<I, Error>
+    where
+        Self: Sized,
+    {
         I::read_from(self)
     }
 }
 
+assert_obj_safe!(Read<'static>);
+
+impl<'a, R: Read<'a> + ?Sized> Read<'a> for &'_ mut R {
+    #[inline]
+    fn read_bytes(&mut self, n: usize) -> Result<&'a [u8], Error> {
+        R::read_bytes(*self, n)
+    }
+
+    #[inline]
+    fn remaining_data(&self) -> usize {
+        R::remaining_data(*self)
+    }
```

Comment:
Don't you want an implementation of `read_le` which uses `R::read_le` rather than using `Self::read_bytes`?

Reply:
Inlining should make this a non-issue, since all those functions are basically trivial. If we see a problem with code bloat, we have ways to deal with it.

Reply:
Yeah, this isn't about inlining, this is about method resolution. :+1: